python/lsst/meas/base/readCalibratedExposure.py

```diff
@@ -0,0 +1,203 @@
+        default="jointcal",
+        allowed={
+            "jointcal": "Use jointcal_wcs"
+        },
+    )
+
+
+class ReadCalibratedExposureTask(Task):
+    """Read and apply calibrations to a calexp
+
+    This is used to apply calibrations from jointcal or fgcm to a calexp.
+
+    If config.doApplyExternalPhotoCalib is `True`, the photometric calibration
+    (`photoCalib`) is taken from `config.externalPhotoCalibName` via the
+    `name_photoCalib` dataset.  Otherwise, the photometric calibration is
+    retrieved from the processed exposure.  When
+    `config.doApplyExternalSkyWcs` is `True`, the astrometric calibration
+    is taken from `config.externalSkyWcsName` with the `name_wcs` dataset.
+    Otherwise, the astrometric calibration is taken from the processed
+    exposure.
+    """
+    ConfigClass = ReadCalibratedExposureConfig
+
+    def readExposure(self, dataRef, calexpType="calexp", bgSubtracted=False):
+        """Read a calexp, optionally restoring the background
+
+        Parameters
+        ----------
+        dataRef : `lsst.daf.persistence.ButlerDataRef`
+            Butler data reference.
+        calexpType : `str`, optional
+            Butler dataset type to retrieve.
+        bgSubtracted : `bool`, optional
+            Return the exposure with the background-subtracted? If ``False``,
+            the ``calexpBackground`` dataset will be used to restore the
+            background.
+
+        Returns
+        -------
+        exposure : `lsst.afw.image.Exposure`
+            Exposure read from disk.
+
+        Raises
+        ------
+        MissingExposureError
+            If data for the exposure is not available.
+        """
+        try:
+            exposure = dataRef.get(calexpType, immediate=True)
+        except NoResults as ex:
+            raise MissingExposureError(f"Exposure not found: {ex}") from ex
+
+        if not bgSubtracted:
+            background = dataRef.get("calexpBackground", immediate=True)
+            exposure.maskedImage += background.getImage()
+
+        return exposure
+
+    def readCalibs(self, dataRef):
+        """Read the photometric and astrometric calibrations for a calexp
+
+        Parameters
+        ----------
+        dataRef : `lsst.daf.persistence.ButlerDataRef`
+            Butler data reference.
+
+        Returns
+        -------
+        calibs : `lsst.pipe.base.Struct`
+            Resultant calibrations, with attributes:
+
+            ``photoCalib``
+                Photometric calibration (`lsst.afw.image.PhotoCalib`), or
+                ``None`` if not ``doApplyExternalPhotoCalib``.
+            ``skyWcs``
+                Astrometric calibration (`lsst.afw.geom.SkyWcs`), or ``None``
+                if not doApplyExternalSkyWcs``.
+        """
+        photoCalib = None
+        if self.config.doApplyExternalPhotoCalib:
+            photoCalib = dataRef.get(f"{self.config.externalPhotoCalibName}_photoCalib")
+
```

Comment:
I think this needs to send back the exposure's {{photoCalib}} even if we aren't applying an external one, or else we don't actually scale the exposure properly below.